File: sol/1233/1233.cpp
struct Solution {
    vector<string> removeSubfolders(vector<string>& folder) {
        vector<string> res;
        ranges::sort(folder);
        string buff;
        for (string &p : folder){
            if (res.empty() || !p.starts_with(buff)){
                res.push_back(p);
                buff = p + '/';
            }
        }
        return res;
    }
};

File: sol/1233/1233.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort()
        res = []
        for p in folder:
            if len(res) == 0 or not p.startswith(res[-1] + '/'):
                res.append(p)
        return res
